### ISSUE TITLE
dotnet workload install takes an incredible amount of time after updating the sdk

### ISSUE THREAD

**Issue by mattleibow** [dotnet/sdk#21319]
### Describe the bug
After updating the .NET SDK from **6.0.100-rc.2.21465.13** to **6.0.100-rc.2.21471.56** I have noticed that the install time goes from 2 mins to 50 mins. 

Noticed in this PR: https://github.com/dotnet/maui/pull/2659

Fast: https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=5242167&view=logs&j=a2b1c795-2687-5fd3-b418-4e4b87c7d580&t=4e29a67b-3a84-5881-9e49-cccb88db0007
Slow: https://dev.azure.com/devdiv/DevDiv/_build/results?buildId=5242060&view=logs&j=78462016-6fc7-57ab-3e4b-3ba758f547e0&t=77e01010-9782-5688-7fce-13f236f3210c

### Further technical details
- Include the output of `dotnet --info`
- The IDE (VS / VS Code/ VS4Mac) you're running on, and its version

**Comment by jonathanpeppers** [comment]
Here is the diff from https://github.com/dotnet/maui/pull/2659: 

https://github.com/dotnet/sdk/compare/608c77bf1451ee87374541a1b4b04d3bfa9abd66...0fae780070ea511a5113aa069107272ded921349

Could it be something about the new retry behavior? /cc @sfoslund @dsplaisted 

I could also reproduce this locally, but `--verbosity diag` output doesn't seem to print exactly what is happening.

**Comment by sfoslund** [comment]
Yeah, the most likely cause in that diff is probably the retry logic. What does `--verbosity diag` print? Do you see the same package being downloaded multiple times?

**Comment by mattleibow** [comment]
I think so:

```
  Installing pack Microsoft.NETCore.App.Runtime.Mono.android-arm64 version 6.0.0-rc.2.21470.23...
  [NuGet Manager] [Info]   GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/registrations2-semver2/microsoft.netcore.app.runtime.mono.android-arm64/index.json
  [NuGet Manager] [Info]   GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/45bacae2-5efb-47c8-91e5-8ec20c22b4f8/nuget/v3/registrations2-semver2/microsoft.netcore.app.runtime.mono.android-arm64/index.json
  [NuGet Manager] [Info]   GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/825db618-e3eb-4426-ba54-b1d6e6c944d8/nuget/v3/registrations2-semver2/microsoft.netcore.app.runtime.mono.android-arm64/index.json
  [NuGet Manager] [Info]   NotFound https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/1a5f89f6-d8da-4080-b15f-242650c914a8/nuget/v3/registrations2-semver2/microsoft.netcore.app.runtime.mono.android-arm64/index.json 66ms
  [NuGet Manager] [Info]   NotFound https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/45bacae2-5efb-47c8-91e5-8ec20c22b4f8/nuget/v3/registrations2-semver2/microsoft.netcore.app.runtime.mono.android-arm64/index.json 70ms
  [NuGet Manager] [Info]   OK https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/825db618-e3eb-4426-ba54-b1d6e6c944d8/nuget/v3/registrations2-semver2/microsoft.netcore.app.runtime.mono.android-arm64/index.json 405ms
  [NuGet Manager] [Info]   GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/825db618-e3eb-4426-ba54-b1d6e6c944d8/nuget/v3/flat2/microsoft.netcore.app.runtime.mono.android-arm64/index.json
  [NuGet Manager] [Info]   OK https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/825db618-e3eb-4426-ba54-b1d6e6c944d8/nuget/v3/flat2/microsoft.netcore.app.runtime.mono.android-arm64/index.json 94ms
  [NuGet Manager] [Info]   GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/825db618-e3eb-4426-ba54-b1d6e6c944d8/nuget/v3/flat2/microsoft.netcore.app.runtime.mono.android-arm64/6.0.0-rc.2.21470.23/microsoft.netcore.app.runtime.mono.android-arm64.6.0.0-rc.2.21470.23.nupkg
  [NuGet Manager] [Info]   OK https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/825db618-e3eb-4426-ba54-b1d6e6c944d8/nuget/v3/flat2/microsoft.netcore.app.runtime.mono.android-arm64/6.0.0-rc.2.21470.23/microsoft.netcore.app.runtime.mono.android-arm64.6.0.0-rc.2.21470.23.nupkg 60ms
  [NuGet Manager] [Info]   GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/825db618-e3eb-4426-ba54-b1d6e6c944d8/nuget/v3/flat2/microsoft.netcore.app.runtime.mono.android-arm64/6.0.0-rc.2.21470.23/microsoft.netcore.app.runtime.mono.android-arm64.6.0.0-rc.2.21470.23.nupkg
  [NuGet Manager] [Info]   OK https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/825db618-e3eb-4426-ba54-b1d6e6c944d8/nuget/v3/flat2/microsoft.netcore.app.runtime.mono.android-arm64/6.0.0-rc.2.21470.23/microsoft.netcore.app.runtime.mono.android-arm64.6.0.0-rc.2.21470.23.nupkg 49ms
  [NuGet Manager] [Info]   GET https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/825db618-e3eb-4426-ba54-b1d6e6c944d8/nuget/v3/flat2/microsoft.netcore.app.runtime.mono.android-arm64/6.0.0-rc.2.21470.23/microsoft.netcore.app.runtime.mono.android-arm64.6.0.0-rc.2.21470.23.nupkg
  [NuGet Manager] [Info]   OK https://pkgs.dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_packaging/825db618-e3eb-4426-ba54-b1d6e6c944d8/nuget/v3/flat2/microsoft.netcore.app.runtime.mono.android-arm64/6.0.0-rc.2.21470.23/microsoft.netcore.app.runtime.mono.android-arm64.6.0.0-rc.2.21470.23.nupkg 65ms
  Writing workload pack installation record for Microsoft.NETCore.App.Runtime.Mono.android-arm64 version 6.0.0-rc.2.21470.23...
```

**Comment by sfoslund** [comment]
Hmm, that actually looks like a normal nuget call, so I don't think it's retrying there. I'll try to repro locally and investigate later today.

**Comment by sfoslund** [comment]
Thanks for reporting this, I made a stupid mistake in the original PR, sorry about that :) Fix here: https://github.com/dotnet/sdk/pull/21329

**Comment by sfoslund** [comment]
Closing as fixed by #21329
